sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp

```diff
@@ -0,0 +1,2630 @@
+        std::string EncryptionKeySHA256;
+        std::string EncryptionAlgorithm;
+      }; // struct StageBlockOptions
+
+      static Azure::Core::Http::Request StageBlockConstructRequest(
+          const std::string& url,
+          const StageBlockOptions& options)
+      {
+        auto request = Azure::Core::Http::Request(
+            Azure::Core::Http::HttpMethod::Put, url, *options.BodyBuffer);
+        request.AddHeader("Content-Length", std::to_string(options.BodyBuffer->size()));
+        request.AddQueryParameter("comp", "block");
+        request.AddQueryParameter("blockid", options.BlockId);
+        request.AddHeader("x-ms-version", options.Version);
+        request.AddHeader("x-ms-date", options.Date);
+        if (!options.ContentMD5.empty())
+        {
+          request.AddHeader("Content-MD5", options.ContentMD5);
+        }
+        if (!options.ContentCRC64.empty())
+        {
+          request.AddHeader("x-ms-content-crc64", options.ContentCRC64);
+        }
+        if (!options.LeaseId.empty())
+        {
+          request.AddHeader("x-ms-lease-id", options.LeaseId);
+        }
+        if (!options.EncryptionKey.empty())
+        {
+          request.AddHeader("x-ms-encryption-key", options.EncryptionKey);
+        }
+        if (!options.EncryptionKeySHA256.empty())
+        {
+          request.AddHeader("x-ms-encryption-key-sha256", options.EncryptionKeySHA256);
+        }
+        if (!options.EncryptionAlgorithm.empty())
+        {
+          request.AddHeader("x-ms-encryption-algorithm", options.EncryptionAlgorithm);
+        }
+        return request;
+      }
+
+      static BlockInfo StageBlockParseResponse(Azure::Core::Http::Response& http_response)
+      {
+        BlockInfo response;
+        auto http_status_code
+            = static_cast<std::underlying_type<Azure::Core::Http::HttpStatusCode>::type>(
+                http_response.GetStatusCode());
+        if (!(http_status_code == 201))
+        {
+          throw std::runtime_error("HTTP status code " + std::to_string(http_status_code));
+        }
+        response.Version = http_response.GetHeaders().at("x-ms-version");
+        response.Date = http_response.GetHeaders().at("Date");
+        response.RequestId = http_response.GetHeaders().at("x-ms-request-id");
+        auto response_clientrequestid_iterator
+            = http_response.GetHeaders().find("x-ms-client-request-id");
+        if (response_clientrequestid_iterator != http_response.GetHeaders().end())
+        {
+          response.ClientRequestId = response_clientrequestid_iterator->second;
+        }
+        auto response_contentmd5_iterator = http_response.GetHeaders().find("Content-MD5");
+        if (response_contentmd5_iterator != http_response.GetHeaders().end())
+        {
+          response.ContentMD5 = response_contentmd5_iterator->second;
+        }
+        auto response_contentcrc64_iterator = http_response.GetHeaders().find("x-ms-content-crc64");
```

Comment:
>x-ms-content-crc64 [](start = 79, length = 18)

Minor: Header key values should be better defined as constant, customer might access them from the raw response. In this case, a constant value is more readable/maintainable than the plain text.

Reply:
Yes, i totally agree. but it is not priority.